core/common/src/main/java/alluxio/heartbeat/FixedIntervalSupplier.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.heartbeat;
+
+public class FixedIntervalSupplier implements SleepIntervalSupplier {
+  private final long mInterval;
+
+  public FixedIntervalSupplier(long fixedInterval) {
+    mInterval = fixedInterval;
+  }
+
+  @Override
+  public long getNextInterval(long mPreviousTickedMs, long nowTimeStampMillis) {
+    long executionTimeMs = nowTimeStampMillis - mPreviousTickedMs;
+    if (executionTimeMs > mInterval) {
+      return 0;
+    }
+    return mInterval;
+  }
+
+  @Override
+  public long getRunLimit(long mPreviousTickedMs) {
+    return Long.MAX_VALUE;
```

Comment:
Shouldn't it return `mInterval`?

Reply:
Not sure what expected behavior of the original SleepTimer? I guess it want to run the heartbeat no limit even exceed the interval, so that the next period heartbeat will not sleep.